addons/mail/models/mail_thread.py

```diff
@@ -2151,19 +2167,33 @@ def _notify_record_by_inbox(self, message, recipients_data, msg_vals=False, **kw
                 bus_notifications.append((self.env['res.partner'].browse(partner_id), 'mail.message/inbox', dict(message_format_values)))
         self.env['bus.bus'].sudo()._sendmany(bus_notifications)
 
-    def _notify_record_by_email(self, message, recipients_data, msg_vals=False,
-                                model_description=False, mail_auto_delete=True, check_existing=False,
-                                force_send=True, send_after_commit=True,
+    def _notify_thread_by_email(self, message, recipients_data, msg_vals=False,
+                                mail_auto_delete=True,  # mail.mail
+                                model_description=False,  # rendering
+                                check_existing=False, force_send=True, send_after_commit=True,  # email send
                                 **kwargs):
         """ Method to send email linked to notified messages.
 
-        :param message: mail.message record to notify;
-        :param recipients_data: see ``_notify_thread``;
-        :param msg_vals: see ``_notify_thread``;
+        :param message: ``mail.message`` record to notify;
+        :param recipients_data: list of recipients information (based on res.partner
+          records), formatted like
+            [{'active': partner.active;
+              'id': id of the res.partner being recipient to notify;
+              'groups': res.group IDs if linked to a user;
+              'notif': 'inbox', 'email', 'sms' (SMS App);
+              'share': partner.partner_share;
+              'type': 'customer', 'portal', 'user;'
+             }, {...}].
+          See ``MailThread._notify_get_recipients``;
+        :param msg_vals: dictionary of values used to create the message. If given
```

Comment:
A little vague, from what I can see it's not used "to create the message".
Or not as I understood it at first when reading (I thought those values would be used to create a record).

From what I gather, it would be more "dictionary of values used as a replacement of the "message" parameter (mail.message record) values".

In the next part, it says "used instead of accessing ``self``", but again, from what I can see, it's used to avoid accessing "message", more than "self", unless I missed something?